reactor-netty-core/src/main/java/reactor/netty/transport/TransportConfig.java

```diff
@@ -169,7 +169,7 @@ public final LoopResources loopResources() {
 	 */
 	@Nullable
 	public final Supplier<? extends ChannelMetricsRecorder> metricsRecorder() {
-		return this.metricsRecorder;
+		return this.metricsRecorderSupplier;
```

Comment:
what is the goal of this method? imagine the `Supplier` initially provided creates a new `ChannelMetricsRecorder` on each call to `get()`. Do we want this method to return such a `Supplier`, or on the contrary an artificial supplier that always returns the current `metricsRecorder` (keeping the `Supplier` return type purely for backward compatibility)?

Reply:
yeah it is for backwards compatibility ... should we deprecate it?

Reply:
yeah, probably... although this brings the question of naming the replacement method.
another thing that might need a bit of rework is the setter (will comment there)

Reply:
Changed it to return a `Supplier` with the resolved metrics recorder